data_structures/src/superblock.rs

```diff
@@ -592,33 +593,21 @@ where
     let each = v.len() / size;
 
     let mut step_index = 0_usize;
-    let mut step = rand_distribution[step_index] as usize;
+    let mut step = rand_distribution[step_index] as usize % each;
     let mut a = first;
     let mut b = (a + step) % v.len();
     while hs_subset.len() < size {
-        if !hs_subset.contains(&v[b]) {
-            hs_subset.insert(v[b].clone());
+        hs_subset.insert(v[b].clone());
```

Comment:
It seems to me that, under the assumption that `v` contains no repetitions, this algorithm can no longer produce collisions because of `% each`, so probably the data structure doesn't need to be a `HashSet` anymore, and we can probably go with something lighter, right?

Furthermore, if that's true, I think we can simply iterate `[0 → size)` instead of trying to procedurally grow the `hs_subset` accumulator up to a target length, because we have guarantee that every iteration will insert exactly one entry into it.